src/controllers/trainee /Controller.ts

```diff
@@ -1,88 +1,75 @@
-import { Request , Response , NextFunction } from 'express';
+import { Request, Response, NextFunction } from 'express';
+import UserRepository from '../../repositories/user/UserRepository';
+import SystemResponse from '../../libs/SystemResponse';
+import IRequest from '../../libs/routes/IRequest';
+import * as bcrypt from 'bcrypt';
 
 class TraineeController {
-static instance ;
-static getInstance = () => {
-if ( TraineeController.instance) {
-return TraineeController.instance;
+  static instance: TraineeController;
+  static userRepository: UserRepository;
+  userRepository = new UserRepository();
+  static getInstance = () => {
+    if (TraineeController.instance) {
+      return TraineeController.instance;
+    }
+    TraineeController.instance = new TraineeController();
+    return TraineeController.instance;
+  }
+  create = async (req: IRequest, res: Response, next: NextFunction) => {
+    try {
+      console.log('::::::::Create Trainee USER:::::::::::::::');
+      const { email, name, address, hobbies, dob, mobileNumber, role, password } = req.body;
```

Comment:
Never destructure this big objects. Use interface to store them in a new variable.